app/views/messages/_message.html.erb

```diff
@@ -46,8 +46,19 @@
       <% end %>
 
       <%# This div needs to be tightly wrapped around the content_text output because the whitespace formatting is important. %>
-      <div class="whitespace-pre-wrap" data-role="content-text" data-clipboard-target="text"
-        ><%= message.content_text %><div class="animate-breathe w-3 h-3 rounded-full bg-black dark:bg-white inline-block ml-1 <%= !thinking && 'hidden' %>"></div></div>
+      <div
+        data-role="content-text"
+        data-clipboard-target="text"
+        class="prose break-words whitespace-pre-wrap leading-normal"
+      ><%= formatted_text = format(message.content_text) %><div
+        class="
+          animate-breathe
+          w-3 h-3
+          rounded-full
+          bg-black dark:bg-white
+          inline-block
+          ml-1 <%= !thinking && 'hidden' %>"></div
+      ></div>
```

Comment:
I had the messy formatting because this div had whitespace-pre-wrap so all line breaks within it were being rendered to the user. However, now that all markdown is converted to HTML I no longer need the pre-wrap. I've cleaned this up.